include/net/xia_ether.h

```diff
@@ -88,6 +108,138 @@ static inline void ether_interface_put(struct ether_interface *eint)
 		interface_finish_destroy(eint);
 }
 
+/* L2 address. */
+struct interface_addr {
+	struct fib_xid_ether_main   *mfxid;
+	struct list_head            interface_common_addr;
+	struct net_device           *outgress_interface;
+	struct rcu_head             rcu_head;
+	u8                          ha[MAX_ADDR_LEN];
+};
+
+/* Main-FXID. */
+struct fib_xid_ether_main {
+	struct xip_dst_anchor   xem_anchor;
+	struct interface_addr   *neigh_addr;
```

Comment:
Isn't it true that if there's a `struct fib_xid_ether_main`, there is a `struct interface_addr` associated with it? That is, isn't there a one to one association here? If so, why do we need `struct interface_addr`?

If we can combine `struct interface_addr` into `struct fib_xid_ether_main`, the code would be simpler by:
1. Dropping redundant fields of `struct interface_addr`, namely, `mfxid` and `rcu_head`.
2. Avoiding memory management code associated with field `rcu_head` of `struct interface_addr`.

Moreover, aren't fields `outgress_interface` and `host_interface` of `struct interface_addr` into `struct fib_xid_ether_main` supposed to always be equal? If so, we can drop one more redundant field of `struct interface_addr`.

Reply:
What you just said is true and i thought of doing the same to make my code simpler, but my reasons for doing so was that 
1. Both ha(coz of alignment issues as stated in hid principal) and fib_xid have the requirement to be last member of a struct and 
2. There could be a situation that somehow externally the interface address associated with an fxid is wrong and so if a request was sent to edit the table then instead of reforming a new fxid we could just edit the interface address associated with it.
I understand now that it is redundant code and will edit my code as soon as possible.